subsys/logging/log_core.c

```diff
@@ -990,4 +993,16 @@ static int enable_logger(void)
 	return 0;
 }
 
+
+void log_flush(void)
+{
+	if (proc_tid) {
+		k_thread_suspend(proc_tid);
+		k_sem_reset(&log_process_done);
+		k_sem_give(&log_process_thread_sem);
+		k_thread_resume(proc_tid);
```

Comment:
I don't know how `k_thread_resume` behaves. If there are no pending logs and logging thread is forever blocked on `(void)k_sem_take(&log_process_thread_sem, timeout);` will it resume that thread so it is able to give the semaphore?

Some time ago I've added flushing in the ztest framework: https://github.com/zephyrproject-rtos/zephyr/blob/c3466b14d09b1e51b6ca472f3b3654e40cba12d8/subsys/testsuite/ztest/src/ztest.c#L939
https://github.com/zephyrproject-rtos/zephyr/blob/c3466b14d09b1e51b6ca472f3b3654e40cba12d8/subsys/testsuite/ztest/src/ztest.c#L722
Can you replace it there with the `log_flush()` you've added?